src/graph_ir/low_graph_ir.h

```diff
@@ -0,0 +1,60 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+/*
+ * This is a small low-level IR that contains particles with flattened 
+ * input/output access path lists. It can be translated directly into datalog
+ * relations that similarly represent particles with flattened lists of full 
+ * access path names.
+ *
+ * There is likely a higher-level IR from which this compiles, and the higher 
+ * level IR can be generated by deserializing protobufs.
+ */
+
+
+#include <string>
```

Comment:
It won't make any practical difference for standard headers that are certainly doing the right thing, but put your `#include`s in the `#ifndef` guard. By placing them here, you're assuming that these headers are themselves `#ifndef`ed correctly. If one of these headers omitted or messed up its #ifndef` guard, you might introduce a very weird and hard-to-debug error by having these dropped in in a case where the rest of this body was omitted due to the `#ifndef` guard.